src/sagemaker/model.py

```diff
@@ -428,18 +431,28 @@ def prepare_container_def(self, instance_type, accelerator_type=None):  # pylint
         deploy_env.update(self._framework_env_vars())
         return sagemaker.container_def(self.image, self.model_data, deploy_env)
 
-    def _upload_code(self, key_prefix):
+    def _upload_code(self, key_prefix, repack=False):
         local_code = utils.get_config_value('local.local_code', self.sagemaker_session.config)
         if self.sagemaker_session.local_mode and local_code:
             self.uploaded_code = None
         else:
-            bucket = self.bucket or self.sagemaker_session.default_bucket()
-            self.uploaded_code = fw_utils.tar_and_upload_dir(session=self.sagemaker_session.boto_session,
-                                                             bucket=bucket,
-                                                             s3_key_prefix=key_prefix,
-                                                             script=self.entry_point,
-                                                             directory=self.source_dir,
-                                                             dependencies=self.dependencies)
+            if repack:
```

Comment:
seems like this part of the conditional should also include `bucket = self.bucket or self.sagemaker_session.default_bucket()` and pass that along to `repack_model()`?

Reply:
out of scope

Reply:
(talked offline.) fair enough. let's track it internally as a possible enhancement though.

Reply:
https://github.com/aws/sagemaker-python-sdk/pull/810